### PR TITLE
[systemtest] Change labels to use strimzi.io/kind instead of type

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Small refactor

### Description

In this PR I'm gonna remove all the additional labels like `type: kafka-mirror-maker` and change all to use `strimzi.io/kind` label for listing certain resource and its pods.

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java [hunk 1]
```diff
@@ -229,13 +226,9 @@ void testDiscoveryAnnotation() {
...
 
-        KafkaBridgeResource.kafkaBridge(bridgeName, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
-            .editMetadata()
-                .addToLabels("type", "kafka-bridge")
-            .endMetadata()
-            .done();
+        KafkaBridgeResource.kafkaBridge(bridgeName, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1).done();
 
-        List<String> bridgePods = kubeClient().listPodNames("type", "kafka-bridge");
+        List<String> bridgePods = kubeClient().listPodNames("type", KafkaBridge.RESOURCE_KIND);
```
Comment (michalxo) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java:231]: Isto to je spravne?
    public static final String RESOURCE_KIND = "KafkaBridge"; != ("type", "kafka-bridge")

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java [hunk 2]
```diff
@@ -234,7 +231,7 @@ void testSecretsWithKafkaConnectS2IWithTlsAndScramShaAuthentication() {
             .withKafkaUsername(userName)
             .build();
 
-        String kafkaConnectS2IPodName = kubeClient().listPods("type", "kafka-connect-s2i").get(0).getMetadata().getName();
+        String kafkaConnectS2IPodName = kubeClient().listPods("type", KafkaConnectS2I.RESOURCE_KIND).get(0).getMetadata().getName();
```
Comment (michalxo) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:234]: `listPods("type", "kafka-connect-s2i")` vs `String RESOURCE_KIND = "KafkaConnectS2I";`
je to ok?

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java [hunk 3]
```diff
@@ -48,6 +48,7 @@
 import static io.strimzi.systemtest.Constants.CONNECT_COMPONENTS;
 import static io.strimzi.systemtest.Constants.MIRROR_MAKER2;
 import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.enums.CustomResourceStatus.Ready;
```
Comment (michalxo) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:51]: Ten import je potrebny?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeST.java:231]: discussed offline, misunderstanding

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java (hunk 2 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:234]: ano je, podivej se kde se to vyuziva

**Comment by im-konge** [comment]
@scholzj @ppatierno I converted this PR to draft as I want to use `strimzi.io/kind` label instead of `type`. And thanks for the reviews.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=HttpBridgeST,ConnectS2IST,ConnectST,KafkaST,MirrorMaker2ST,MirrorMakerST,SecurityST,OauthPlainST,OauthTlsST,TracingST,AllNamespaceST

**Comment by michalxo** [comment]
LGTM

**Comment by samuel-hawker** [comment]
I _thought_ that using strimzi labels on custom resources was banned (and causes the operator to throw exceptions) would be interested to see if this works...

**Comment by scholzj** [comment]
@samuel-hawker But he is not setting them on CRs, or? He is just using those set by the operator.

**Comment by samuel-hawker** [comment]
Apologies, I reviewed the last iteration of the PR where he added the labels to the CR and misread this review and thought he was switching those labels, not removing them!
You're abosolutely right, this works well!

**Comment by im-konge** [comment]
Yeah sorry, I changed it to grab the `strimzi.io/kind` label and not use additional (and almost the same) labels like `type: kafka-mirror-maker`.

**Comment by samuel-hawker** [comment]
No need to apologize, it was me who misread ;P

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=io.strimzi.systemtest.bridge.HttpBridgeST#testScaleBridgeToZero,io.strimzi.systemtest.bridge.HttpBridgeST#testScaleBridgeSubresource

**Pull request merged**